src/components/recentlyViewed/RecentlyViewed.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useEffect, useState } from 'react';
+import { getProductDetail } from '../../data/API';
+import style from './RecentlyViewed.module.css';
+
+const RecentlyViewed = () => {
+  const [products, setProducts] = useState([]);
+  const watchedProducts = JSON.parse(localStorage.getItem('watched'));
+  useEffect(() => {
+    // for(let i = 0; i < watchedProducts.length; i++) {
+    //   let details = getProductDetail(watchedProducts[i])
+    //   details.then((data) => {
+    //     setProducts((curArr)=>[data, ...curArr])
+    //   });
+    // }
+    if (watchedProducts) {
+      for (let products of watchedProducts) {
+        let details = getProductDetail(products);
+        details.then((data) => {
+          setProducts((curArr) => [data, ...curArr]);
```

Comment:
```suggestion
          setProducts((prevProducts) => [data, ...prevProducts]);
```
변수명만으로 무엇을 담고 있는지 나타내면 좋습니다